extensions/pkg/controller/utils.go

```diff
@@ -240,6 +241,7 @@ func tryUpdate(ctx context.Context, backoff wait.Backoff, c client.Client, obj r
 
 		if err := updateFunc(ctx, obj); err != nil {
 			if apierrors.IsConflict(err) {
+				reflect.ValueOf(obj).Elem().Set(reflect.ValueOf(resetCopy).Elem())
```

Comment:
I was also wondering the same thing today.
But after this copy we again `Get` the object from the server into `obj` which should update the resourceVersion. Or am I missing something.
Also /thumbsup for the unit test.

Reply:
> But after this "copy", we again Get the object from the server into obj which should update the resourceVersion. Or am I missing something.

I think this has ever happened since we introduced this tryUpdate procedure, hasn't it? In case of a conflict we always tried to refetch the object and thus overwrite the necessary fields of the very same referenced object.

Reply:
Yes, I used this https://gist.github.com/swilen-iwanow/9dee546b30cfc41e509b79a38692f511 to test it. 
Concurrently updating the status and verifying that the state is there, after a conflict has occurred. 

Reseting the object help us to start from the beginning and clear the transformation we did to the object which later messes the equality check.

Prior to this in case of a conflict we fetch the object but don't clear the transformation of the object so it is a "dirty" version of it because of the previous iteration. And after we transform the refetched "dirty" object and compare it to the new transformed they are equal and we return.

Reply:
> In case of a conflict we always tried to refetch the object and thus overwrite the necessary fields of the very same referenced object.

Ah, of course. We must have missed that.
Nevertheless, @rfranzke had some concerns regarding the reflection stuff. Can we neglect them or should we go with the safer approach (with schemes)?

> Yes, I used this https://gist.github.com/swilen-iwanow/9dee546b30cfc41e509b79a38692f511 to test it.
Concurrently updating the status and verifying that the state is there, after a conflict has occurred.

Perfect. Can you transform that into a unit test and add it to this PR?

Reply:
@timebertt If the solution as it is currently works, should we then rather go for it instead of changing the functions signature with the scheme? The main argument would be that it's easier to use (although I still don't like reflection in general, but that's just a personal preference, so if it works as desired with this one additional line I can live with it 😉)

Reply:
Sounds good to me :)
Then let's go for this approach 🚀

Reply:
The only think I'd love to see @swilen-iwanow is a short comment about why this is necessary, since we discussed so much in this thread 😄